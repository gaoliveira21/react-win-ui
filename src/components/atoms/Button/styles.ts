import styled, { css, DefaultTheme } from 'styled-components'

import { ButtonProps } from '.'

const modifiers = {
  primary: (theme: DefaultTheme) => css`
    color: ${theme.colors.text.ON_ACCENT};
    background-color: ${theme.colors.fill.accent.ACCENT_PRIMARY};

    &:hover {
      background-color: ${theme.colors.fill.accent.ACCENT_SECONDARY};
    }

    &:active {
      background-color: ${theme.colors.fill.accent.ACCENT_TERTIARY};
    }

    &:disabled {
      color: ${theme.colors.text.ON_ACCENT_DISABLED};
      background-color: ${theme.colors.fill.accent.DISABLED};
    }
  `,

  secondary: (theme: DefaultTheme) => css`
    color: ${theme.colors.text.PRIMARY};
    background-color: ${theme.colors.fill.control.CONTROL_PRIMARY};

    &:hover {
      background-color: ${theme.colors.fill.control.CONTROL_SECONDARY};
    }

    &:active {
      background-color: ${theme.colors.fill.control.CONTROL_TERTIARY};
    }

    &:disabled {
      color: ${theme.colors.text.DISABLED};
      background-color: ${theme.colors.fill.control.CONTROL_PRIMARY};
    }
  `
}

export const Button = styled.button<Pick<ButtonProps, 'variant'>>`
  ${({ theme, variant }) => css`
    font-size: ${theme.fonts.text.BODY.size};
    font-weight: ${theme.fonts.text.BODY.weight};
    line-height: ${theme.fonts.text.BODY.lineHeight};

    box-shadow: 0 0 2px 2px ${theme.colors.elevation.control.BORDER};

    padding: 5px 47px;
    border-radius: 4px;

    transition: background-color 0.3s;

    &:disabled {
      cursor: not-allowed;
    }

    &:focus {
      outline: 3px solid ${theme.colors.stroke.focus.OUTER};
      border: 1px solid ${theme.colors.stroke.focus.INNER};
    }

    ${!!variant && modifiers[variant](theme)};
  `}
`
