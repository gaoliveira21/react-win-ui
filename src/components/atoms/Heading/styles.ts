import styled, { css } from 'styled-components'

import { HeadingProps } from '.'

export const Heading = styled.div<Pick<HeadingProps, 'variant' | 'color'>>`
  ${({ theme, variant, color }) => css`
  font-family: ${theme.fonts.heading[variant.toUpperCase()].fontFamily};}
    font-size: ${theme.fonts.heading[variant.toUpperCase()].size};}
    font-weight: ${theme.fonts.heading[variant.toUpperCase()].weight};}
    line-height: ${theme.fonts.heading[variant.toUpperCase()].lineHeight};}
    color: ${theme.colors.text[color.toUpperCase()]};
  `}
`
