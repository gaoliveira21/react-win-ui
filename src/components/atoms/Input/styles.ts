import styled, { css } from 'styled-components'

export const Input = styled.input`
  ${({ theme }) => css`
    background-color: ${theme.colors.fill.control.CONTROL_PRIMARY};
    border: 1px solid ${theme.colors.elevation.textControl.BORDER};
    border-bottom: 2px solid ${theme.colors.stroke.control.SECONDARY};

    color: ${theme.colors.text.PRIMARY};
    font-size: ${theme.fonts.text.BODY.size};
    font-weight: ${theme.fonts.text.BODY.weight};
    line-height: ${theme.fonts.text.BODY.lineHeight};

    padding: 4px 35px 6px 11px;
    border-radius: 3px;

    &::placeholder {
      font-size: ${theme.fonts.text.BODY.size};
      font-weight: ${theme.fonts.text.BODY.weight};
      line-height: ${theme.fonts.text.BODY.lineHeight};
      color: ${theme.colors.text.SECONDARY};
    }

    &:focus {
      outline: none;
      border-bottom: 2px solid ${theme.colors.fill.accent.ACCENT_PRIMARY};
      background-color: ${theme.colors.fill.control.ACTIVE};
    }

    &:hover {
      &:not(:focus, :disabled) {
        background-color: ${theme.colors.fill.control.CONTROL_SECONDARY};
      }
    }

    &:disabled {
      background-color: ${theme.colors.fill.control.DISABLED};
      color: ${theme.colors.text.DISABLED};
      cursor: not-allowed;
    }
  `}
`
