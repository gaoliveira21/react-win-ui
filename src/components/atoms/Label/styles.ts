import styled, { css } from 'styled-components'

export const Label = styled.label`
  ${({ theme }) => css`
    font-size: ${theme.fonts.text.BODY.size};
    font-weight: ${theme.fonts.text.BODY.weight};
    line-height: ${theme.fonts.text.BODY.lineHeight};
    color: ${theme.colors.text.PRIMARY};
  `}
`
