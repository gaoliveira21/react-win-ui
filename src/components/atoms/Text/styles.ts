import styled from 'styled-components'

import { TextProps } from '.'

export const Text = styled.p<Pick<TextProps, 'variant'>>`
  font-family: ${({ theme, variant }) =>
    theme.fonts.text[variant.toUpperCase()].fontFamily};
  font-size: ${({ theme, variant }) =>
    theme.fonts.text[variant.toUpperCase()].size};
  font-weight: ${({ theme, variant }) =>
    theme.fonts.text[variant.toUpperCase()].weight};
  line-height: ${({ theme, variant }) =>
    theme.fonts.text[variant.toUpperCase()].lineHeight};
`
