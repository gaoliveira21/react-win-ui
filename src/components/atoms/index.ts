export * from './Text'
export * from './Heading'
export * from './Button'
export * from './Input'
export * from './Label'
