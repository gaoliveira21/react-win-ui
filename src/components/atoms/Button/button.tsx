import * as S from './styles'

export type ButtonProps = {
  children: React.ReactNode
  disabled?: boolean
  onClick?: React.MouseEventHandler<HTMLButtonElement>
  variant?: 'primary' | 'secondary'
}

export function Button({
  children,
  disabled = false,
  onClick,
  variant = 'primary'
}: ButtonProps): React.ReactElement {
  return (
    <S.Button variant={variant} onClick={onClick} disabled={disabled}>
      {children}
    </S.Button>
  )
}
