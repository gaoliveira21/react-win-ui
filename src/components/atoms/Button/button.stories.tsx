import { Meta, Story } from '@storybook/react/types-6-0'

import { Button } from './button'

export default {
  title: 'Button',
  component: Button
} as Meta

export const Basic: Story = (args) => <Button {...args}>Button</Button>
