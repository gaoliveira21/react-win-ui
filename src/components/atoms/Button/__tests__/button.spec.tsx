import { screen } from '@testing-library/react'

import { Button } from '@/components/atoms/Button'
import { renderWithTheme } from '@/tests/render-with-theme'
import { lightAccentFillColors } from '@/styles/colors'

describe('<Button />', () => {
  it('Should render button with correct text content', () => {
    renderWithTheme(<Button>text</Button>)

    expect(screen.getByRole('button', { name: /text/i })).toBeInTheDocument()
  })

  it('Should apply the correct styles if disabled', () => {
    renderWithTheme(<Button disabled>text</Button>)

    const button = screen.getByRole('button', { name: /text/i })

    expect(button).toBeDisabled()
    expect(button).toHaveStyleRule(
      'background-color',
      lightAccentFillColors.DISABLED,
      { modifier: ':disabled' }
    )
  })
})
