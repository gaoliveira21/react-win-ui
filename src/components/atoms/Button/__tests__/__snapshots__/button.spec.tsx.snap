// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render button with correct text content 1`] = `
.c0 {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  box-shadow: 0 0 2px 2px rgba(0,0,0,0.06);
  padding: 5px 47px;
  border-radius: 4px;
  -webkit-transition: background-color 0.3s;
  transition: background-color 0.3s;
  color: rgb(255,255,255);
  background-color: rgb(0,95,184);
}

.c0:disabled {
  cursor: not-allowed;
}

.c0:focus {
  outline: 3px solid rgba(0,0,0,0.9);
  border: 1px solid rgb(255,255,255);
}

.c0:hover {
  background-color: rgba(0,95,184,0.9);
}

.c0:active {
  background-color: rgba(0,95,184,0.8);
}

.c0:disabled {
  color: rgb(255,255,255);
  background-color: rgba(0,0,0,0.22);
}

<button
  class="c0"
>
  text
</button>
`;
